packages/mini-oxygen/src/vite/server-middleware.ts

```diff
@@ -207,20 +201,59 @@ export function setupOxygenMiddleware(
     },
   );
 
+  let miniOxygen: MiniOxygen;
+  let miniOxygenOptions: MiniOxygenViteOptions;
+
   viteDevServer.middlewares.use(function o2HandleWorkerRequest(req, res) {
     // This request comes from the browser. At this point, Vite
     // tried to serve the request as a static file, but it didn't
     // find it in the project. Therefore, we assume this is a
     // request for a backend route, and we forward it to workerd.
 
-    dispatchFetch(toWeb(req))
-      .then((webResponse) => pipeFromWeb(webResponse, res))
-      .catch((error) => {
-        console.error('Error during evaluation:', error);
-        res.writeHead(500);
-        res.end();
-      });
+    const ready =
+      miniOxygen && !miniOxygen.isDisposed
+        ? Promise.resolve()
+        : getMiniOxygenOptions().then((options) => {
+            miniOxygenOptions = options;
+            miniOxygen = startMiniOxygenRuntime(options);
+          });
```

Comment:
Create or restart the instance if it has been disposed (when running `viteServer.close()`, which happens on `vite.config.js` save automatically).